plutus-core/plutus-core/src/PlutusCore/TypeCheck/Internal.hs

```diff
@@ -139,13 +160,18 @@ lookupBuiltinM ann fun = do
         Just ty -> liftDupable ty
 
 -- | Extend the context of a 'TypeCheckM' computation with a typed variable.
-withVar :: Name -> Normalized (Type TyName uni ()) -> TypeCheckM uni fun cfg err a -> TypeCheckM uni fun cfg err a
+withVar
+    :: Monad m
+    => Name
+    -> Normalized (Type TyName uni ())
+    -> TypeCheckT uni fun cfg m a
+    -> TypeCheckT uni fun cfg m a
 withVar name = local . over tceVarTypes . insertByName name . dupable
```

Comment:
>  Doesn't matter here unless you really hate that `Monad` constraint.

I do (because it makes me ask "why is it here?"), thank you, I wasn't aware of that.